chore(provider): SailPoint Identity Secure Cloud (ISC) (#10723)
nextauthjs/next-auth: docs/pages/getting-started/providers/sailpoint.mdx

      profile(profile: IdentitySecureCloudProfile) {
        return {
            tenant: profile.tenant,
            id: profile.id,
            uid: profile.uid,
            email: profile.email,
            phone: profile.phone,
            workPhone: profile.workPhone,
            firstname: profile.firstname,
            lastname: profile.lastname,
            capabilities: profile.capabilities,
            displayName: profile.displayName,
            name: profile.uid,
        }
      },
      style: { text: "#011E69", bg: "#fff", logo: "sailpoint.svg" },
    },
  ],
})
```

  </Code.Svelte>
  <Code.Express>

```ts filename="/src/app.ts"
import { ExpressAuth } from "@auth/express"

app.use("/auth/*", ExpressAuth({ providers: [
  {
      id: "identitySecureCloud",
      name: "Identity Secure Cloud",
      type: "oauth",
      clientId: process.env.ISC_CLIENT_ID!,
      clientSecret: process.env.ISC_CLIENT_SECRET!,
      authorization: {
        url: `${process.env.ISC_BASE_URL!}/oauth/authorize`,
        params: { scope: 'sp:scopes:all' },
      },
      token: `${process.env.ISC_BASE_API_URL!}/oauth/token`,
      userinfo: `${process.env.ISC_BASE_API_URL!}/oauth/userinfo`,

>       profile(profile: IdentitySecureCloudProfile) {

Reviewer: This should be somethign like this, for all 3 examples

```ts
      profile(profile: IdentitySecureCloudProfile) {
        return {
            id: profile.id,
            email: profile.email,
            name: `${profile.firstname} ${profile.lastname}`,
            image: null
        }
      },
```

Author: Thanks a lot for the suggestion, already made the changes, I still keep name: profile.uid. The uid value is like yu-chih.chung, which is more useful as the login account name for SailPoint Identity Secure Cloud. I think display name is less usefuly in this case.

        return {
            tenant: profile.tenant,
            id: profile.id,
            uid: profile.uid,
            email: profile.email,
            phone: profile.phone,
            workPhone: profile.workPhone,
            firstname: profile.firstname,
            lastname: profile.lastname,
            capabilities: profile.capabilities,
            displayName: profile.displayName,
            name: profile.uid,
        }
      },
      style: { text: "#011E69", bg: "#fff", logo: "sailpoint.svg" },
    },
] }))
```

  </Code.Express>

> </Code>

Reviewer: And somethign like this below the example code to point folks in the right direction, should they want to use any of the additional fields

```suggestion
</Code>

Your `userprofile` endpoint will return more fields, but by default the [User table](https://authjs.dev/getting-started/database#models) only supports `id`, `name`, `email`, and `image`. Therefore, if you'd like to use any of the following fields, make sure you modify the `User` table schema in whichever adapter / database you're using.

	```ts
	tenant: profile.tenant,
	id: profile.id,
	uid: profile.uid,
	email: profile.email,
	phone: profile.phone,
	workPhone: profile.workPhone,
	firstname: profile.firstname,
	lastname: profile.lastname,
	capabilities: profile.capabilities,
	displayName: profile.displayName,
	name: profile.uid
	```

The above fields will all be available in the `profile` callback.

```

Author: Thanks a lot, this looks perfect, in my case, I also did the same which extend the User Type and also User table schema

import { Callout } from "nextra/components"
import { Code } from "@/components/Code"

<img align="right" src="/img/providers/sailpoint.svg" height="64" width="64" />

# SailPoint Identity Secure Cloud Provider

> SailPoint Identity Secure Cloud (ISC) is an enterprise SaaS platform for identity/security solution. In order to test with this OAuth integration, you will need an ISC tenant, if your are a sailpoint customer or partner, you may inquiry with SailPoint partner/customer success manager. If you are a developer, you may inquiry more information on [SailPoint Developer Community](https://developer.sailpoint.com/discuss/) for more information. 

Reviewer: ```suggestion
SailPoint Identity Secure Cloud (ISC) is an enterprise SaaS platform for identity and security. In order to use this OAuth integration, you will need an ISC tenant. If you're a SailPoint customer or partner, please talk to your SailPoint account manager for more details. If you are a developer, you can check out the [SailPoint Developer Community](https://developer.sailpoint.com/discuss/). 
```

## Resources

- [SailPoint Identity Secure Cloud Authentication](https://developer.sailpoint.com/docs/api/authentication#choose-authorization-grant-flow)
- [Managing API Keys and Personal Access Tokens](https://documentation.sailpoint.com/saas/help/common/api_keys.html?h=oauth+client#creating-an-api-key)
- [SailPoint Developer Community](https://developer.sailpoint.com/discuss/)

## Setup

### Callback URL

<Code>
  <Code.Next>

```bash
https://example.com/api/auth/callback/identitySecureCloud
```

  </Code.Next>
  <Code.Svelte>

```bash

> https://example.com/api/auth/callback/identitySecureCloud

Reviewer: In all other packages the default dynamic route is at `/auth/*` not `/api/auth/*`, small difference.

```suggestion
https://example.com/auth/callback/identitySecureCloud
```

```

  </Code.Svelte>
</Code>

### Create OAuth Client

> Find your Identity Secure Cloud Tenant OAuth Information (can be retrieved under https://{tenant}.api.identitynow.com/oauth/info). Create OAuth Client ([guide](https://documentation.sailpoint.com/saas/help/common/api_keys.html?h=oauth+client#creating-an-api-key))with grant types: AUTHORIZATION_TOKEN and REFRESH_TOKEN, Redirect URL: Callback URL above,   select scopes: sp:scope:all. Note down the clientId and clientSecret.

Reviewer: ```suggestion
Find your Identity Secure Cloud Tenant OAuth Information which can be found at `https://{tenant}.api.identitynow.com/oauth/info`. Create an OAuth Client (following this [guide](https://documentation.sailpoint.com/saas/help/common/api_keys.html?h=oauth+client#creating-an-api-key)) with grant types: `AUTHORIZATION_TOKEN` and `REFRESH_TOKEN`. Redirect URL should match your version of the Callback URL above. Finally, select the scopes `sp:scope:all`. Note down the generated `clientId` and `clientSecret`.
```

      authorization: {
        url: `${process.env.ISC_BASE_URL!}/oauth/authorize`,
        params: { scope: 'sp:scopes:all' },
      },
      token: `${process.env.ISC_BASE_API_URL!}/oauth/token`,
      userinfo: `${process.env.ISC_BASE_API_URL!}/oauth/userinfo`,
      profile(profile: IdentitySecureCloudProfile) {
        return {
            tenant: profile.tenant,
            id: profile.id,
            uid: profile.uid,
            email: profile.email,
            phone: profile.phone,
            workPhone: profile.workPhone,
            firstname: profile.firstname,
            lastname: profile.lastname,
            capabilities: profile.capabilities,
            displayName: profile.displayName,
            name: profile.uid,
        }
      },
      style: { text: "#011E69", bg: "#fff", logo: "sailpoint.svg" },
    },
  ],
})
```

  </Code.Next>
  <Code.Svelte>

```ts filename="/src/auth.ts"
import { SvelteKitAuth } from "@auth/sveltekit"
import SailPoint from "@auth/sveltekit/providers/sailpoint"

export const { handle, signIn, signOut } = SvelteKitAuth({
  providers: [
    {
      id: "identitySecureCloud",
      name: "Identity Secure Cloud",
      type: "oauth",

>       clientId: process.env.ISC_CLIENT_ID!,

Reviewer: The sveltekit environment variables work a bit different, check out this example:

![image](https://github.com/nextauthjs/next-auth/assets/7415984/77d18b83-e358-4ead-876b-8335ff1c8a03)

Author: Hello, 

Thanks a lot for the correction. Just made the change in the PR.

```ts filename="/src/app.ts"
import { ExpressAuth } from "@auth/express"

app.use("/auth/*", ExpressAuth({ providers: [
  {
      id: "identitySecureCloud",
      name: "Identity Secure Cloud",
      type: "oauth",
      clientId: process.env.ISC_CLIENT_ID!,
      clientSecret: process.env.ISC_CLIENT_SECRET!,
      authorization: {
        url: `${process.env.ISC_BASE_URL!}/oauth/authorize`,
        params: { scope: 'sp:scopes:all' },
      },
      token: `${process.env.ISC_BASE_API_URL!}/oauth/token`,
      userinfo: `${process.env.ISC_BASE_API_URL!}/oauth/userinfo`,
      profile(profile: IdentitySecureCloudProfile) {
        return {
            tenant: profile.tenant,
            id: profile.id,
            uid: profile.uid,
            email: profile.email,
            phone: profile.phone,
            workPhone: profile.workPhone,
            firstname: profile.firstname,
            lastname: profile.lastname,
            capabilities: profile.capabilities,
            displayName: profile.displayName,
            name: profile.uid,
        }
      },
      style: { text: "#011E69", bg: "#fff", logo: "sailpoint.svg" },
    },
] }))
```

  </Code.Express>
</Code>

> ### Configuration for Refresh Token

Reviewer: Is there anything about this refresh process that's different with SailPoint? If not, lets drop this section or just link to the existing guide. If there are significant differences, then lets point those out.

Author: I think we can drop this section off first, gernally it's the same as the others, was just thinking provider some example there.

Author: Just removed it. Thanks.
